qubesbuilder/plugins/source/__init__.py

```diff
@@ -0,0 +1,163 @@
+                "--keys-dir", str(PLUGINS_DIR / "source/keys")
+            ]
+            for maintainer in self.component.maintainers:
+                get_sources_cmd += ["--maintainer", maintainer]
+            if self.component.insecure_skip_checking:
+                get_sources_cmd += ["--insecure-skip-checking"]
+            if self.component.less_secure_signed_commits_sufficient:
+                get_sources_cmd += ["--less-secure-signed-commits-sufficient"]
+            bash_cmd = [
+                f"cd {str(BUILDER_DIR)}", " ".join(get_sources_cmd)
+            ]
+            cmd = [
+                "/bin/bash", "-c", "&&".join(bash_cmd)
+            ]
+            self.executor.run(cmd, copy_in, copy_out)
+
+            # Update parameters based on previously fetched sources as .qubesbuilder
+            # is now available.
+            self.update_parameters()
+
+            distfiles_dir = self.get_distfiles_dir()
+            distfiles_dir.mkdir(parents=True, exist_ok=True)
+
+            # Download and verify files given in .qubesbuilder
+            for file in self.parameters.get("files", []):
+                fn = os.path.basename(file["url"])
+                copy_in = [
+                    (self.plugins_dir / "source", PLUGINS_DIR),
+                    (self.component.source_dir, BUILDER_DIR),
+                ]
+                copy_out = [
+                    (source_dir / fn, distfiles_dir)
+                ]
+                # Build command for "download-and-verify-file". We let the script checking
+                # necessary options.
+                download_verify_cmd = [
+                    str(PLUGINS_DIR / "source/scripts/download-and-verify-file"),
+                    "--output-dir", str(BUILDER_DIR / self.component.name),
+                    "--file-url", file['url'],
+                ]
+                if file.get("sha256", None):
+                    download_verify_cmd += [
+                        "--checksum-cmd", "sha256sum", "--checksum-file",
+                        str(BUILDER_DIR / self.component.name / file['sha256']),
+                    ]
+                elif file.get("sha512", None):
+                    download_verify_cmd += [
+                        "--checksum-cmd", "sha512sum", "--checksum-file",
+                        str(BUILDER_DIR / self.component.name / file['sha512']),
+                    ]
+                if file.get("signature", None):
+                    download_verify_cmd += ["--signature-url", file['signature']]
+                if file.get("pubkey", None):
+                    download_verify_cmd += ["--pubkey-file", file['pubkey']]
+                bash_cmd = [
+                    f"cd {str(BUILDER_DIR / self.component.name)}",
+                    " ".join(download_verify_cmd)
+                ]
+                cmd = [
+                    "/bin/bash", "-c", "&&".join(bash_cmd)
+                ]
+                self.executor.run(cmd, copy_in, copy_out)
```

Comment:
Please use subprocess for this, instead of dynamically generating a shell command.